docs/model_developers/modules.rst

```diff
@@ -122,3 +122,96 @@ The format for requires in requirements.txt is the folllowing:
+Python packages, you can double check the desired modules are installed in editable mode by checking
+the output of ``pip list --editable``.
+
+
+Working on the dev environment
+------------------------------
+After setting up, you should be left with a dev environment where all required v2 modules have been
+installed (either in editable or in packaged form). If you're working on a project, all required v1
+modules should be checked out in the ``modulepath`` directory.
+
+When you run a compile from the active Python environment context, the compiler will find both the
+v1 and v2 modules and use them for both their model and their plugins.
+
+Similarly, when you run a module's unit tests, the installed v2 modules will automatically be used
+by the compiler. As for v1 modules, by default, the ``pytest-inmanta`` extension makes sure the
+compile itself runs against an isolated project, downloading any v1 module dependencies at
+compile-time. If you want to compile against local versions of v1 modules, have a look at the
+``--use-module-in-place`` option in the ``pytest-inmanta`` documentation.
+
+
+Distributing modules
+====================
+This section is about v2 modules. V1 modules only require a version tag to be recognized as a
+released version. While a version tag is still good practice for v2 modules, it isn't sufficient
+to consider it released.
+
+You can package a v2 module with ``inmanta module build`` which will create an sdist and a bdist
+of the Python package. You can then publish this to the Python package repository of your choice,
+for example the public PyPi repository. For an inmanta project, follow the same procedure but
+substitute ``module`` with ``project``.
+
+The orchestrator server generally (see
+:ref:`Advanced concepts<modules-distribution-advanced-concepts`) installs both project and modules
+from the configured Python package repository, respecting the environment's version constraints on
+the project package, the project's constraints on its modules and all inter-module constraints. The
+server is then responsible for supplying the agents with the appropriate ``inmanta_plugins``
+modules.
+
+.. _modules-distribution-advanced-concepts
+
+Advanced concepts
+-----------------
+
+Freezing a project
+^^^^^^^^^^^^^^^^^^
+Prior to releasing a new stable version of an inmanta project, you might wish to freeze its module
+dependencies. This will ensure that the orchestrator server will always work with the exact
+versions specified. You can achieve this with
+``inmanta project freeze --recursive --operator "=="``. This command will freeze all module
+depdencies to their exact version as they currently exist in the Python environment. The recursive
+option makes sure all module depdencies are frozen, not just the direct dependencies. In other
+words, if the project depdends on module ``a`` which in turn depends on module ``b``, both modules
+will be pinned to their current version in ``pyproject.toml``.
```

Comment:
```suggestion
``inmanta project freeze --recursive --operator "=="``. This command will freeze all module
dependencies to their exact version as they currently exist in the Python environment. The recursive
option makes sure all module dependencies are frozen, not just the direct dependencies. In other
words, if the project depends on module ``a`` which in turn depends on module ``b``, both modules
will be pinned to their current version in ``pyproject.toml``.
```

Reply:
Wow, this really shows an issue with my typing...